src/lib/blueprint.ts

```diff
@@ -1,15 +1,90 @@
 import type { Openapi } from './openapi.js'
 
+interface Parameter {
+  name: string
+  isRequired: boolean
+  description: string
+}
+
+interface Response {
+  description: string
+}
+
+type Method = 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH'
+
+interface Endpoint {
+  name: string
+  path: string
+  method: Method[]
+  semanticMethod?: string
+  routeDescription?: string | undefined
+  parameters: Parameter[]
+  response: Response
+}
+
+interface Route {
+  name: string
+  path: string
+  namespace: Namespace | null
+  endpoints: Endpoint[]
+  subroutes: Route[] | null
+}
+
+interface Namespace {
+  name: string
+  path: string
+}
+
 export interface Blueprint {
```

Comment:
When defining the types, it's much easier read this code if we start at the top and define the nodes below. E.g., instead of

	
```
	interface Parameter {}
	interface Endpoint {}
	interface Route {}
	interface Namespace {}
	interface Blueprint {}
	```

Prefer this order


```
	interface Blueprint {}
	interface Route {}
	interface Namespace {}
	interface Endpoint {}
	interface Parameter {}
```